src/Main.elm

```diff
@@ -36,15 +37,45 @@ type alias Model =
     }
 
 
+startGame : Difficulty -> Model
+startGame difficulty =
+    { initialModel
+        | game = difficulty
+        , grid = Grid.fromDimensions (getDimensions difficulty)
+    }
+
+
+initialModel : Model
+initialModel =
+    { grid = Grid.fromDimensions (getDimensions initialDifficulty)
+    , activeCell = Nothing
+    , pressingFace = False
+    , game = initialDifficulty
+    , time = 0
+    , mode = Start
+    , isRightClicked = False
+    , menu = Nothing
+    }
+
+
 type Difficulty
     = Beginner
     | Intermediate
     | Expert
-    | Custom Int Int Int
+    | Custom GridProperties
```

Comment:
i wanted to keep this type pretty much as-is. i pulled out `GridProperties` to be more explicit with names, rather than keeping track of the order of 3 `Int`s, and because it comes in handy when making the `Menu`.